app/views/targets/index.json.jbuilder

```diff
@@ -1,9 +1,16 @@
 json.array! @all_targets do |target|
-  json.set! :topic,     target.topic
-  json.set! :length,    target.length
-  json.set! :title,     target.title
-  json.set! :latitude,  target.latitude
-  json.set! :longitude, target.longitude
-  json.set! :user,      target.user.id
-  json.set! :edit_url,  edit_target_url(target)
+  json.set! :topic,        target.topic
+  json.set! :length,      target.length
+  json.set! :title,       target.title
+  json.set! :latitude,    target.latitude
+  json.set! :longitude,   target.longitude
+  json.set! :user,        target.user.id
+  if current_user == target.user
+    json.set! :edit_url,  edit_target_url(target)
+  end
+  if @matches.include? target
```

Comment:
all this block can be reduced to
`json.set! :compatible, @matches.include? target`